### PR TITLE
Use our own RegTest network server

### PR DESCRIPTION
**Pull request by dcousens**
Woo!

> 38 passing (9s)

For integration tests.
No faucet issues! (we hope, ha)

### CODE REVIEW

File: test/integration/_regtest.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
+var bitcoin = require('../../')
+var dhttp = require('dhttp/200')
+
+var APIPASS = process.env.APIPASS || 'satoshi'
```
Comment (dcousens) [inline test/integration/_regtest.js]: If this is actually an issue (doubt it),  I'll make this private.

**Comment by junderw** [comment]
wooooooooot! I'll check it out right now

File: test/integration/cltv.js [hunk 2]
```diff
@@ -34,36 +34,100 @@ describe('bitcoinjs-lib (transactions w/ CLTV)', function () {
...
+        tx.setInputScript(0, redeemScriptSig)
+
+        regtestUtils.broadcast(tx.toHex(), function (err) {
+          // fails before the expiry
+          assert.throws(function () {
+            if (err) throw err
+          }, /Error: 64: non-final/)
+
+          // into the future!
+          regtestUtils.mine(51, function (err) {
```
Comment (dcousens) [inline test/integration/cltv.js]: @junderw isn't this great? :)

**Comment by dcousens** [comment]
@junderw server code can be found at https://github.com/bitcoinjs/private-bitcoin

File: test/integration/_mainnet.js [hunk 3]
```diff
@@ -1,3 +0,0 @@
-var Blockchain = require('cb-http-client')
-var BLOCKTRAIL_API_KEY = process.env.BLOCKTRAIL_API_KEY || 'c0bd8155c66e3fb148bb1664adc1e4dacd872548'
-module.exports = new Blockchain('https://api.blocktrail.com/cb/v0.2.1/BTC', { api_key: BLOCKTRAIL_API_KEY })
```
Comment (dcousens) [inline test/integration/_mainnet.js]: @rubensayshi happy? :smiley:

**Comment by dabura667** [comment]
Huzzah!

**Pull request merged**